### PR TITLE
highway: add patch to fix build issue with RPi4 and enable shared libs

### PR DESCRIPTION
**Pull request by clopez**
* The patch was upstreamed at https://github.com/google/highway/pull/1589

### CODE REVIEW

File: recipes-extended/highway/files/pr_1589.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+and that failed to build when it tried to inline code from other parts that by
+default is compiled with `-march=armv8-a+crc` (due to the toolchain in this
+example).
+
+The solution is to append the crypto extension, so in this example of the RPi4 we
+end compiling HWY_NEON with `-march=armv8-a+crc+crypto` allowing it to inline code
+from any other part that is built with `-march=armv8-a+crc`.
+See the documentation from GCC for the `Inlining rules` in the links from above.
+
+Issues related: #1570 and #1460.
```
Comment (psaavedra) [inline recipes-extended/highway/files/pr_1589.patch:34]:
```suggestion
Issues related: #1570 and #1460.

Upstream-Status: Accepted [https://github.com/google/highway/pull/1589]
```

**Review by psaavedra — APPROVED** [review]
I'd add just the small suggestion in the patch that follows the 
[OE Commit Patch Message Guidelines](https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines)

File: recipes-extended/highway/files/pr_1589.patch (hunk 1 above)
Comment (clopez) [reply recipes-extended/highway/files/pr_1589.patch:34]: good suggestion, i will apply it

**Comment by clopez** [comment]
merged as 42ebac8357cbff4f5992c0ec2217889ac276ca32

**Pull request closed without merging**